### ISSUE TITLE
[java] Parse error with local class with 2 or more annotations

### ISSUE THREAD

**Issue by scais** [pmd/pmd#208]
Hello guys,

   I found another issue when analyzing open-source project 'spring' (https://github.com/spring-projects/spring-framework). Can you please take a look?

NOTE: The problems seems to be the custom annotation `@Configuration`, after its removal the analyzer does not complain. 

**PMD version:**
5.5.2

**Rule Set:**
java-basic

**Description:**
```
Error while processing file: spring-framework\spring-context\src\test\java\org\springframework\context\annotation\configuration\BeanAnnotationAttributePropagationTests.java
net.sourceforge.pmd.lang.java.ast.ParseException: Encountered "" at line 121, column 17.
Was expecting one of:

        at net.sourceforge.pmd.lang.java.ast.JavaParser.generateParseException(JavaParser.java:10875)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.jj_consume_token(JavaParser.java:10759)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.BlockStatement(JavaParser.java:4629)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.Block(JavaParser.java:4575)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.MethodDeclaration(JavaParser.java:1554)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.ClassOrInterfaceBodyDeclaration(JavaParser.java:1246)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.ClassOrInterfaceBody(JavaParser.java:1162)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.ClassOrInterfaceDeclaration(JavaParser.java:586)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.TypeDeclaration(JavaParser.java:476)
        at net.sourceforge.pmd.lang.java.ast.JavaParser.CompilationUnit(JavaParser.java:222)
        at net.sourceforge.pmd.lang.java.AbstractJavaParser.parse(AbstractJavaParser.java:56)
        at net.sourceforge.pmd.SourceCodeProcessor.parse(SourceCodeProcessor.java:95)
        at net.sourceforge.pmd.SourceCodeProcessor.processSource(SourceCodeProcessor.java:142)
        at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:80)
        at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:47)
        at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:79)
        at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:25)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

```

**Code Sample demonstrating the issue:**

```
	@Test
	public void defaultLazyConfigurationPropagatesToIndividualBeans() {
		@Lazy @Configuration class Config {
			@Bean Object foo() { return null; }
		}

		assertTrue("@Bean methods declared in a @Lazy @Configuration should be lazily instantiated",
				beanDef(Config.class).isLazyInit());
	}
```
**Running PMD through:** *[CLI]*
```
java -cp * net.sourceforge.pmd.PMD -d "spring-framework\spring-web-reactive\src\main\java\org\springframewor
k\web\reactive\function" -f text -R java-basic -verbose
```

**Comment by jsotuyod** [comment]
@scais thanks again for the report. You running PMD over several different open source libraries has certainly been useful. We should probably select a few and run an analysis ourselves when running from Travis at the very least.

I confirm the issue, the grammar currently supports only 1 annotation for local classes.

I'm working on a fix at the moment.
